pyat/at/lattice/lattice_object.py

```diff
@@ -436,6 +455,71 @@ def particle(self, particle: Union[str, Particle]):
         else:
             self._particle = Particle(particle)
 
+    def set_fillpattern(self, bunches: Union[int, float,
+                                             numpy.ndarray] = 1):
+        """Function to generate the filling pattern lof the ring.
+        The filling pattern is computed as:
+
+        ``bunches/numpy.sum(bunches)``
+
+        This function also generates the bunch spatial distribution
+        accessible with ``Lattice.bunch_spos``
+
+        Keyword Arguments:
+           bunches:  scalar or array of positive double or bool to define
+                     the bunch distribution.
+                     For scalar input, equidistant bunches are assumed.
+                     ``ring.harmonic_number`` has to be a multiple of
+                     ``bunches``.
+                     For array input the condition
+                     ``len(bunches)==ring.harmonic_number`` is required.
+                     (default=1, single bunch configuration).
+        """
+        if bunches == 1:
+            fp = numpy.ones(1)
+        elif numpy.isscalar(bunches):
```

Comment:
The special case `bunches == 1` does not create an array with the right size. You may take out this special case, `numpy.isscalar(1)` should work.

Reply:
`numpy.ones(1)` is an accepted valid array, but I agree it is not useful, I have removed this functionality